src/MCP23017.cpp

```diff
@@ -22,12 +22,16 @@ void MCP23017::init()
 	writeRegister(MCP23017_REGISTER::GPPUA, 0xFF, 0xFF);
 }
 
-void MCP23017::portMode(MCP23017_PORT port, uint8_t value)
+void MCP23017::portMode(MCP23017_PORT port, uint8_t value, bool inverted)
 {
 	writeRegister(MCP23017_REGISTER::IODIRA + port, value);
+
+	if( inverted ){
+		writeRegister(MCP23017_REGISTER::IPOLA + port, value);
+	}
 }

```

Comment:
I think another `uint8_t` parameter instead of a `bool` for inverted would make more sense.
For one, one cannot set pin in different polarities at once with this signature, which defeats the purpose of it. Moreover, with the current code, the polarity cannot be unset.

```suggestion
void MCP23017::portMode(MCP23017_PORT port, uint8_t dir, uint8_t inverted)
{
   writeRegister(MCP23017_REGISTER::IODIRA + port, dir);
   writeRegister(MCP23017_REGISTER::IPOLA + port, inverted);
}
```